tasks/upload_finisher.py

```diff
@@ -154,71 +159,85 @@ def run_impl(
-                    # working properly.
-                    parallel_verification_task.apply_async(
-                        kwargs=dict(
+                    log.info(
+                        "Saving combined report",
+                        extra=dict(
                             repoid=repoid,
-                            commitid=commitid,
-                            commit_yaml=commit_yaml,
-                            report_code=report_code,
-                            parallel_paths=parallel_paths,
+                            commit=commitid,
                             processing_results=processing_results,
+                            parent_task=self.request.parent_id,
                         ),
                     )
 
+                    if parallel_processing is ParallelProcessing.PARALLEL:
+                        pr = processing_results["processings_so_far"][0][
+                            "arguments"
+                        ].get("pr")
+                        processor_task = UploadProcessorTask()
+                        processor_task.save_report_results(
+                            db_session,
+                            report_service,
+                            repository,
+                            commit,
+                            report,
+                            pr,
+                            report_code,
+                        )
+                        state.mark_uploads_as_merged(upload_ids)
+
+                    else:
+                        parallel_paths = report_service.save_parallel_report_to_archive(
+                            commit, report, report_code
+                        )
+                        # now that we've built the report and stored it to GCS, we have what we need to
+                        # compare the results with the current upload pipeline. We end execution of the
+                        # finisher task here so that we don't cause any additional side-effects
+
+                        # The verification task that will compare the results of the serial flow and
+                        # the parallel flow, and log the result to determine if parallel flow is
+                        # working properly.
+                        parallel_verification_task.apply_async(
+                            kwargs=dict(
+                                repoid=repoid,
+                                commitid=commitid,
+                                commit_yaml=commit_yaml,
+                                report_code=report_code,
+                                parallel_paths=parallel_paths,
+                                processing_results=processing_results,
+                            ),
+                        )
+            except LockError:
+                max_retry = 200 * 3**self.request.retries
+                retry_in = min(random.randint(max_retry // 2, max_retry), 60 * 60 * 5)
```

Comment:
So 60*60*5 is 5 hours correct? Are we okay that being the minimum number to retry for big numbers of self.request.retries?

Reply:
So far this was just copied from the other place that does locking. I would like to revisit this at some point, and reduce the retry intervals, as well as use try-lock instead of blocking.